service/src/main/java/bio/terra/cbas/util/BackfillOriginalWorkspaceIdService.java

```diff
@@ -0,0 +1,175 @@
+package bio.terra.cbas.util;
+
+import bio.terra.cbas.config.CbasContextConfiguration;
+import bio.terra.cbas.dao.MethodDao;
+import bio.terra.cbas.dao.MethodVersionDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.models.Method;
+import bio.terra.cbas.models.MethodVersion;
+import bio.terra.cbas.models.RunSet;
+import bio.terra.common.db.WriteTransaction;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/*
+NOTE: This class is inherently temporary.  Once all original workspace IDs have been backfilled,
+      this entire class should be deleted, along with its method calls in BackfillOriginalWorkspaceIdBean.java
+ */
+@Service
+public class BackfillOriginalWorkspaceIdService {
+  private final Logger logger = LoggerFactory.getLogger(BackfillOriginalWorkspaceIdService.class);
+  @Autowired CbasContextConfiguration cbasContextConfig;
+  @Autowired RunSetDao runSetDao;
+  @Autowired MethodDao methodDao;
+  @Autowired MethodVersionDao methodVersionDao;
+
+  public BackfillOriginalWorkspaceIdService(
+      CbasContextConfiguration cbasContextConfig,
+      RunSetDao runSetDao,
+      MethodDao methodDao,
+      MethodVersionDao methodVersionDao) {
+    this.cbasContextConfig = cbasContextConfig;
+    this.methodDao = methodDao;
+    this.runSetDao = runSetDao;
+    this.methodVersionDao = methodVersionDao;
+  }
+
+  @WriteTransaction
+  public void backfillAll() {
+    if (cbasContextConfig
+        .getWorkspaceCreatedDate()
+        .isEqual(OffsetDateTime.parse("0000-01-01T00:00:00.000000Z"))) {
+      logger.info(
+          "Skipping BackfillOriginalWorkspaceId operation. This could indicate that workspaceCreatedDate in application.yml has not been configured properly.");
```

Comment:
> This could indicate that workspaceCreatedDate in application.yml has not been configured properly

What else could it indicate? Config not getting through feels like a error level problem to me (rather than info). Is there any valid situation where this would be left unconfigured?

In the worse case, this feels risky:
- Config is bad
- Backfilling is skipped (instead of failing)
- Cleanup continues and deletes a bunch of content that should have been backfilled

It feels safer (since this is more of a "we always expect it to be configured properly) to error out at this point.

What do you think?

Reply:
you're right... I arranged things this way to get around some problems related to the integration tests, but I think I should find a different way.